packages/next/src/server/app-render/dynamic-rendering.ts

```diff
@@ -652,19 +664,22 @@ function createErrorWithComponentStack(
   return error
 }
 
+export const PRELUDE_FULL = 0
+export const PRELUDE_EMPTY = 1
+export const PRELUDE_ERRORED = 2
+
 export function throwIfDisallowedDynamic(
   workStore: WorkStore,
-  hasEmptyShell: boolean,
+  prelude: 0 | 1 | 2,
   dynamicValidation: DynamicValidationState,
-  serverDynamic: DynamicTrackingState,
-  clientDynamic: DynamicTrackingState
+  serverDynamic: DynamicTrackingState
 ): void {
   if (workStore.invalidDynamicUsageError) {
     console.error(workStore.invalidDynamicUsageError)
     throw new StaticGenBailoutError()
   }
 
-  if (hasEmptyShell) {
+  if (prelude > PRELUDE_FULL) {
```

Comment:
do we need to rely on the ordering of the underlying numeric representation? seems like it'd be more readable to just do this:
```suggestion
  if (prelude !== PRELUDE_FULL) {
```